tensorflow/core/kernels/split_v_op.cc

```diff
@@ -184,6 +184,57 @@ class SplitVOpBase : public OpKernel {
 template <typename T, typename Tlen, typename InputReshapedType, int NDims>
 class SplitVOpCPUImpl {
  public:
+  void ParallelSplitByInputData(OpKernelContext* context,
+                                const InputReshapedType& input_reshaped,
+                                const TensorShape& input_shape,
+                                const std::vector<Tlen>& split_sizes_vec,
+                                const int32 split_dim) const {
+    T* p_data = (T*)input_reshaped.data();
+    const size_t input_elements_num = input_reshaped.size();
+    uint32_t elem_pkg = input_reshaped.dimensions().rank() == 3
+                                  ? input_reshaped.dimension(2)
+                                  : 1;
+    //elem_pkg = 10;
+    const uint32_t line_elem_num =
+        (input_reshaped.dimensions().rank() >= 2 ? input_reshaped.dimension(1)
+                                                 : 1) *
+        elem_pkg;
+    const uint32_t line_num = input_reshaped.dimension(0);
+    T** res = new T*[split_sizes_vec.size()];
+    for (int64 i = 0; i < split_sizes_vec.size(); ++i) {
+      TensorShape output_shape(input_shape);
+      output_shape.set_dim(split_dim, split_sizes_vec[i]);
+      Tensor* result = nullptr;
+      OP_REQUIRES_OK(context,
+                     context->allocate_output(i, output_shape, &result));
+      res[i] = (T*)(&result->flat<T>()(0));
+    }
+
+    int32_t start = 0;
+    auto func = [split_sizes_vec, &p_data, elem_pkg, &res, line_elem_num](
+                    int32_t start_part, int32_t end_part) {
+      int start = start_part * line_elem_num; // 0
+      int end = end_part * line_elem_num; // 930
+      uint32_t times = 0;
+      for (int32_t i = start; i < end;) {
+        for (int32_t j = 0; j < split_sizes_vec.size(); ++j) {
+          const auto jump_elem_num = split_sizes_vec[j] * elem_pkg;
+          memcpy(&(res[j][(start_part + times) * jump_elem_num]), p_data + i,
```

Comment:
Avoid `memcpy`, it leads to undefined behavior for non-trivial types.  `std::copy` or `std::copy_n` are better, and will end up using `memcpy` under-the-hood for trivial types.